guava/src/com/google/common/cache/CacheBuilder.java

```diff
@@ -760,8 +759,28 @@ Ticker getTicker(boolean recordsTime) {
     return this;
   }
 
+  /**
+   * Enable the accumulation of {@link CacheStats} during the operation of the cache. Without this
+   * {@link Cache#stats} will return zero for all statistics. Note that recording stats requires
+   * bookkeeping to be performed with each operation, and thus imposes a performance penalty on
+   * cache operation.
+   *
+   * @param statsCounter a custom stats counter.
+   */
+  public CacheBuilder<K, V> recordStats(final StatsCounter statsCounter) {
```

Comment:
This has to accept a Supplier, as the builder is reusable and a counter is generally per-cache. That's not horrible at the call site due to`Suppliers.ofInstance` or `CustomCounter::new`